pkg/utils/db.go

```diff
@@ -201,26 +202,32 @@ func GetMasterStatus(ctx context.Context, db *sql.DB, flavor string) (gmysql.Pos
 		pos        uint32
 		nullPtr    interface{}
 	)
-	for rows.Next() {
-		if len(rowColumns) == 5 {
-			err = rows.Scan(&binlogName, &pos, &nullPtr, &nullPtr, &gtidStr)
-		} else {
-			err = rows.Scan(&binlogName, &pos, &nullPtr, &nullPtr)
-		}
-		if err != nil {
-			return binlogPos, gs, terror.DBErrorAdapt(err, terror.ErrDBDriverError)
-		}
+	if !rows.Next() {
+		return binlogPos, gs, terror.ErrNoMasterStatus.Generate()
+	}
 
-		binlogPos = gmysql.Position{
-			Name: binlogName,
-			Pos:  pos,
-		}
+	if len(rowColumns) == 5 {
+		err = rows.Scan(&binlogName, &pos, &nullPtr, &nullPtr, &gtidStr)
+	} else {
+		err = rows.Scan(&binlogName, &pos, &nullPtr, &nullPtr)
```

Comment:
Do we need `gtidStr` for mariaDB?
https://github.com/pingcap/dumpling/blob/master/v4/export/metadata.go#L137